NEWS.md

```diff
@@ -415,6 +415,8 @@ Library improvements
     * `mktemp` and `mktempdir` now take an optional argument to set which
       directory the temporary file or directory is created in.
 
+    * New, generational garbage collector ([#5227]). In some use cases it can reduce the worst case time for garbage collection by a factor of ten. Julia can now be used for soft real-time applications like audio processing or high-level robot control.
```

Comment:
I would prefer not to make specific quantitative claims about the GC's performance.